request-controller/src/test/java/org/wildfly/extension/requestcontroller/WFCORE4967_TestCase.java

```diff
@@ -0,0 +1,74 @@
+package org.wildfly.extension.requestcontroller;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+
+public class WFCORE4967_TestCase {
+    private static final int TASKS_QTY = 10;
+    private static final int THREADS_QTY = 20;
+
+    @Test
+    public void noQueuedTasksLossWhenRunningRequestCompleteOnSuspendedRC() throws InterruptedException {
+        final AtomicInteger executedTaskCount = new AtomicInteger();
+
+        RequestController rc = suspendedRCWithQueuedTasks(TASKS_QTY, () -> {
+            executedTaskCount.incrementAndGet();
+        });
+
+        CountDownLatch latch = new CountDownLatch(THREADS_QTY);
+        List<Thread> threads = createSynchronisedThreads(latch, () -> {
+            rc.requestComplete();
+        });
+        threads.forEach(Thread::start);
+        // wait until all above threads ready to fire rc.requestComplete() together with bellow rc.resume()
+        latch.await();
+        rc.resume();
+
+        for (Thread t : threads) {
+            t.join();
+        }
+
+        // simulate just enough requests after server is resumed to drain potential outstanding tasks from taskQueue
+        for (int requestNo = 0; requestNo < TASKS_QTY; requestNo++) {
+            rc.requestComplete();
+        }
+
+        assertEquals(TASKS_QTY, executedTaskCount.intValue());
+    }
+
+    private RequestController suspendedRCWithQueuedTasks(int i, Runnable whenExecuted) {
+        RequestController _ret = new RequestController(false);
+        _ret.suspended(() -> {

```

Comment:
could you please rename all `_ret` variables ? that's the only remaining request before I can merge this PR.
thanks

Reply:
Sure, done.